### DISCUSSION TITLE
Use env variable for jobs.id.runs-on and jobs.id.services.image ?

### DISCUSSION THREAD

**Discussion by abdellani** [community/community#25092] [Actions]
<p>Hi,</p>
<p>I  wrote a workflow to test a rails applications with Rubocop and RSpec. Now, I’m trying to move the values for some parameters like jobs.id.runs-on and jobs.id.services.image to environment variables but It doesn’t work. Github actions complains about “env”:</p>
<pre><code>- Your workflow file was invalid: The pipeline is not valid. .github/workflows/workflow.yml (Line: 14, Col: 14): Unrecognized named-value: 'env'. Located at position 1 within expression: env.OS_VERSION,.github/workflows/workflow.yml (Line: 27, Col: 14): Unrecognized named-value: 'env'. Located at position 1 within expression: env.OS_VERSION
</code></pre>
<p>the source</p>
<p><a href="https://gist.github.com/abdellani/d8146ada301127bedb50ca816da5456c" rel="nofollow noopener">https://gist.github.com/abdellani/d8146ada301127bedb50ca816da5456c</a></p>
<p>Kind regards.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thanks abdellani for reaching this out!</p>
<p>${{ env.OS_VERSION  }} should work but just hasn’t been implemented yet, it’s being tracked internally.</p>
<p>I don’t have any timeline to share yet. Currently please directly use exact os ‘ubuntu-18.04’ for ‘run-on’ on job level.</p>

**Comment by Croydon** [comment]
<p>This is confusing me</p>
<p>I’m trying to do something similar. I’m definining top-level <em>env</em> variables and try to access them in <em>jobs.if</em></p>
<p>From the documentation I assumed that this should already work and I spend now 2h or so trying arbitrary stuff as I ran out of ideas.</p>
<p>So GitHub is still working on implementing support for accessing env varibles in <em>job.run-on</em> and <em>job.if</em> and similar keywords?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/croydon">@croydon</a> , </p>
<p>Yes, for some job level, eg: job.run-on, job.if, job.container, github is working on to support top-level env variables.</p>
<p>For step level, env variables are working well now.</p>

**Comment by lifelofranco** [comment]
2 years in and this still does not work 🥲
